src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class ResolveRuntimePackAssets : TaskBase
+    {
+        public ITaskItem[] ResolvedRuntimePacks { get; set; }
+
+        [Output]
+        public ITaskItem[] RuntimePackAssets { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            List<TaskItem> runtimePackAssets = new List<TaskItem>();
+
+            foreach (var runtimePack in ResolvedRuntimePacks)
+            {
+                string runtimePackRoot = runtimePack.GetMetadata(MetadataKeys.PackageDirectory);
+                string runtimeIdentifier = runtimePack.GetMetadata(MetadataKeys.RuntimeIdentifier);
+
+                string runtimeAssetsPath = Path.Combine(runtimePackRoot, "runtimes", runtimeIdentifier, "lib", "netcoreapp3.0");
+                string nativeAssetsPath = Path.Combine(runtimePackRoot, "runtimes", runtimeIdentifier, "native");
+
+                var runtimeAssets = Directory.Exists(runtimeAssetsPath) ? Directory.GetFiles(runtimeAssetsPath) : Array.Empty<string>();
+                var nativeAssets = Directory.Exists(nativeAssetsPath) ? Directory.GetFiles(nativeAssetsPath) : Array.Empty<string>();
+
+                void AddAsset(string assetPath, string assetType)
+                {
+                    if (Path.GetExtension(assetPath).Equals(".pdb", StringComparison.OrdinalIgnoreCase))
```

Comment:
Not sure this will rate perf-wise in practice, but you can avoid allocating for every file with `assetPath.EndsWith(".pdb", StringComparison.OrdinalIgnoreCase)`. These sorts of things did add up in the optimization of ResolvePackageAssets.